### PR TITLE
CATROID-1237 REFACTOR SensorLoudness to Kotlin

### PR DESCRIPTION
**Pull request by hofcle78**
[CATROID-1237 REFACTOR SensorLoudness to Kotlin](https://jira.catrob.at/browse/CATROID-1237)

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/soundrecorder/SoundRecorder.java [hunk 1]
```diff
@@ -38,12 +38,16 @@
...
+		this("/dev/null");
+	}
+
 	public SoundRecorder(String path) {
 		this.recorder = new MediaRecorder();
 		this.path = path;
 	}
 
-	public void start() throws IOException, RuntimeException {
+	public void start() throws IOException, IllegalStateException {
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/soundrecorder/SoundRecorder.java:50]: This method, is a bit odd, why bother to catch an exception if it's thrown in the same catch clause? 
Please explain your change here :)

File: catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorLoudness.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                recorder = SoundRecorder()
+            }
+            lastValue = 0.0
+        }
+    }
+
+    companion object {
+        private const val UPDATE_INTERVAL = 50
+        private const val SCALE_RANGE = 100.0
+        private const val MAX_AMP_VALUE = 32_767.0
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorLoudness.kt:110]: Refactor to private compile-time constants, please.
https://kotlinlang.org/docs/properties.html#compile-time-constants

File: catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorLoudness.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        private val TAG = SensorLoudness::class.java.simpleName
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/formulaeditor/SensorLoudness.kt:111]: Make it a private "val" of SensorLoudness to get rid of the companion object, please.

File: catroid/src/main/java/org/catrobat/catroid/ui/dialogs/FormulaEditorComputeDialog.java [hunk 4]
```diff
@@ -115,6 +115,7 @@ public void setFormula(Formula formula) {
 
 	@Override
 	protected void onStop() {
+		SensorHandler.stopSensorListeners();
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/dialogs/FormulaEditorComputeDialog.java:118]: Please explain this change :)

**Review by gPathpp — CHANGES_REQUESTED** [review]
Add some Tests, please.

**Comment by reichli** [comment]
Closed due to inactivity. Can be used as reference by the next developer.

**Pull request closed without merging**